### DISCUSSION TITLE
Link tracking

### DISCUSSION THREAD

**Discussion by yolk23321** [openlit/openlit#553] [Q&A]
Does OpenLIT provide detailed link tracking features, such as: query rewriting, querying the RAG, querying the vector database, and returning the final result, with each step displayed?

**Comment by amanagarwal042** [comment]
Could you share a bit more detail on this. We do collect complete execution trace of events that happen but I think you are looking for more than that?

**Reply by yolk23321** [reply]
<img width="675" alt="image" src="https://github.com/user-attachments/assets/90017fc2-0d97-4ac2-bfb3-6ac1519ab2f7" />
In langgrah or Langchain, there are some callback methods. With the code in the image above, does OpenLIT allow you to get all the monitoring logs just by initializing it? I tried it, but it didn’t work

**Comment by yolk23321** [comment]
Also, does simply calling openlit.init() allow for tracking the chain of method calls in nested functions? I’ve tried it, but it doesn’t seem to work.

**Comment by amanagarwal042** [comment]
> With the code in the image above, does OpenLIT allow you to get all the monitoring logs just by initializing it? I tried it, but it didn’t work

Yes it does, When you initialise it does automatically generating traces and metrics. Whats your react code, can you share that?
